### ISSUE TITLE
"division w/o __future__ statement (old-division)" triggers even with future statement

### ISSUE THREAD

**Issue by cumber** [PyCQA/pylint#852]
If you enable this warning, it seems to trigger if you have any division (unless it's obviously float division, such as involving a float literal), regardless of whether or not you've included the `from __future__ import division` it recommends.

```python
# division.py
from __future__ import division
print 1 / 3
```

Running `pylint --reports=no --disable=all --enable=old-division division.py` reports:

```
No config file found, using default configuration
************* Module division
W:  3, 6: division w/o __future__ statement (old-division)
```

pylint --version:
```
pylint 1.5.4, 
astroid 1.4.4
Python 2.7.11 (default, Dec  5 2015, 19:47:20) 
[GCC 4.9.3]
```

(I'm on Nixos, but pylint is installed through pip into a virtualenv, not from Nixos' packages)

**Comment by cumber** [comment]
Might be related to #781, which is also a warning triggering in the presence of `from __future__ import division` when it shouldn't

**Comment by PCManticore** [comment]
By the way, that message should be enabled with the Python 3 porting checker (``--py3k`` flag), most of those patterns don't make sense to have activated by default.

**Comment by cumber** [comment]
Oh, I was hoping to be able to establish a code style rule "division shouldn't be used without from future import __division__", for a Python2 codebase we have no current intention of porting to Python3.

When I found the old-division checker I thought it could be a big help; will it not work properly without some of the the py3k stuff?

**Comment by PCManticore** [comment]
Not necessarily, it might work, but you might hit edge cases such as this one, since the Python 3 porting checkers were written with the purpose of being executed together. Anyway, thank you for discovering this. ;-)
